src/formatDate.js

```diff
@@ -1,14 +1,62 @@
 'use strict';
 
-/**
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateToArrow = date.split(fromFormat[fromFormat.length - 1]);
+  const formatedDate = [];
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'DD':
+        day = dateToArrow[i];
+        break;
+
+      case 'MM':
+        month = dateToArrow[i];
+        break;
+
+      default:
+        year = dateToArrow[i];
+        break;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i]) {
+      case 'DD': {
+        formatedDate.push(day);
+        break;
+      }
+
+      case 'MM': {
+        formatedDate.push(month);
+        break;
+      }
+
+      case 'YY': {
+        if (year.length === 4) {
+          formatedDate.push(year.slice(2));
+        } else {
+          formatedDate.push(year);
+        }
```

Comment:
When converting from 'YYYY' to 'YY', the code correctly slices the last two digits. However, there's no check for the 'YY' to 'YYYY' conversion in this case. If the input year is already in 'YY' format, it should be converted to 'YYYY' according to the specified logic.